src/pages/DelegatoryValidator/UnstakeDialog.tsx

```diff
@@ -5,41 +5,106 @@ import {
   DialogContent,
   DialogTitle,
   FormHelperText,
+  IconButton,
   InputAdornment,
   OutlinedInput,
+  Stack,
+  Typography,
 } from "@mui/material";
-import React from "react";
-import ContentBox from "../../components/IndividualPageContent/ContentBox";
-import ContentRow from "../../components/IndividualPageContent/ContentRow";
+import React, {useState} from "react";
+import CloseIcon from "@mui/icons-material/Close";
+import {grey} from "../../themes/colors/aptosColorPalette";
+import {Types} from "aptos";
+import {parseTimestamp, timestampDisplay} from "../utils";
 
 type isDialogOpen = {
   handleDialogClose: () => void;
   isDialogOpen: boolean;
+  accountResource?: Types.MoveResource | undefined;
 };
 
 export default function UnstakeDialog({
   handleDialogClose,
   isDialogOpen,
+  accountResource,
 }: isDialogOpen) {
+  const lockedUntilSecs = accountResource
+    ? BigInt((accountResource.data as any).locked_until_secs)
+    : null;
+  const moment = parseTimestamp(Number(lockedUntilSecs).toString());
+  const timestamp_display = timestampDisplay(moment);
+
+  const [unstakedAmount, setUnstakedAmount] = useState<string>();
+  const [unstakedAmountError, setUnstakedAmountError] =
+    useState<boolean>(false);
+  const handleUnstakedAmountChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const amt = event.target.value;
+    setUnstakedAmountError(isNaN(Number(amt)));
+    setUnstakedAmount(amt);
+  };
+
   return (
     <Dialog open={isDialogOpen} onClose={handleDialogClose} fullWidth={true}>
-      <DialogTitle>Unstake funds</DialogTitle>
-      <DialogContent>
-        <FormHelperText>Enter amount</FormHelperText>
-        <OutlinedInput
-          placeholder="0"
-          endAdornment={<InputAdornment position="end">APT</InputAdornment>}
-        />
-      </DialogContent>
-      <ContentBox>
-        <ContentRow title="Transaction fee" value={null} />
-      </ContentBox>
-      <DialogActions>
-        <Button onClick={handleDialogClose}>Unstake</Button>
-      </DialogActions>
-      <DialogContent>
-        Your funds will be unlocked after the lock period in ? days.
-      </DialogContent>
+      <IconButton
```

Comment:
the styled dialog (with a close button) is used more than once now. let's give it a proper component and reuse that component.